catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryList.java

```diff
@@ -32,16 +32,25 @@
 	@Override
 	public synchronized boolean add(Object o) {
 		boolean status = false;
-		PointF point = (PointF) o;
-
-		if (super.isEmpty() || isNewStitchPoint(point)) {
+		if (o instanceof PointF) {
+			PointF point = (PointF) o;
+
+			if (super.isEmpty() || isNewStitchPoint(point)) {
+				status = super.add(new StitchPoint(point));
+			}
+		} else if (o instanceof StitchPoint) {
+			StitchPoint point = (StitchPoint) o;
 			status = super.add(point);
 		}
 
 		return status;
 	}
 
 	private Boolean isNewStitchPoint(PointF point) {
```

Comment:
since "StitchPoint" is now a class, isNewStitchPoint is kind of misleading to take PointF as parameter.